src/stream/src/executor/aggregation/agg_impl/approx_distinct_append.rs

```diff
@@ -50,8 +62,14 @@ impl RegisterBucket for AppendOnlyRegisterBucket {
 
 #[derive(Clone, Debug, Default)]
 pub struct AppendOnlyStreamingApproxCountDistinct {
-    // TODO(yuchao): The state may need to be stored in state table to allow correct recovery.
     registers: Vec<AppendOnlyRegisterBucket>,
+
+    /// Group key to aggregate with group.
+    /// None for simple agg, Some for group key of hash agg.
+    group_key: Option<Row>,
```

Comment:
Yes. `MaterializedInputState` can easily get rid of `group_key` after #5908, by using `value_indices` (though not yet implemented in that PR). For new _table state_ like `ApproxCountDistinct` in this PR, I'm not quite sure if we can achieve that, because inserting into state table requires group key.

Reply:
> we should maintain the group key in the cache if we store the group_key here, which will take unnecessary memory cost. if we can pass it as a parameter everywhere, we can save it. The other states have the issue too, so we can also fix it later. c.c @richardchien

fixed